packages/babel-preset-react-app/webpack-overrides.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+'use strict';
+
+const crypto = require('crypto');
+
+module.exports = function() {
+  return {
+    // This function transforms the Babel configuration on a per-file basis
+    config(config, { source }) {
+      // Babel Macros are notoriously hard to cache, so they shouldn't be
+      // https://github.com/babel/babel/issues/8497
+      // We naively detect macros using their package suffix and add a random token
+      // to the caller, a valid option accepted by Babel, to compose a one-time
+      // cacheIdentifier for the file. We cannot tune the loader options on a per
+      // file basis.
+      if (source.indexOf('.macro') !== -1 || source.indexOf('/macro') !== -1) {
```

Comment:
Nit: you can do one lookup for just `macro`.
Then check the character at previous index.

Reply:
To make that work we'd have to save the index into a variable. Is that tradeoff worth it?

Reply:
At that point, we might as well just look for `macro`. If the previous character wasn't `.` or `/`, we'd probably want to search again.

FWIW, `indexOf` appears to be extremely fast: https://jsperf.com/test-indexof-vs-includes
This only needs to be ran once on a file that doesn't include macros.

Reply:
I assume regex would be way slower?

Reply:
>To make that work we'd have to save the index into a variable. Is that tradeoff worth it?

What do you mean? There's no real cost for putting it into a variable in this case.

Reply:
>I assume regex would be way slower?

It depends. Regex can actually be faster.

Reply:
I just mean the cost of both seems quite low so is it worth trading one for the other? I think the way it is currently it's easy to read and understand what's going on. I think it gets a bit more complicated/less clear if we change the algorithm (see @Timer's comment above).

Reply:
@Timer I would assume regex would be slower too but then I started to second guess that. 

https://koukia.ca/top-6-ways-to-search-for-a-string-in-javascript-and-performance-benchmarks-ce3e9b81ad31#---0-21

Maybe doing both checks in one go with a regex would be fastest?

Reply:
I'm fine with using a regex.

I disagree re: the cost being low. That's how you get to slow builds, one small decision at a time. If it's a hot path (which it is for cold builds) let's put in some effort to avoid doing the same work twice.

Reply:
Meh. Whatever. I'm just being grumpy. It probably doesn't matter at this scale. Things like this tend to matter in React but this is not React.